tajo-core/src/main/java/org/apache/tajo/engine/planner/physical/BNLJoinExec.java

```diff
@@ -190,15 +152,12 @@ public Tuple next() throws IOException {
         }
       }
 
-      frameTuple.set(leftTuple, rightIterator.next());
-      if (hasJoinQual) {
-        if (joinQual.eval(inSchema, frameTuple).isTrue()) {
-          projector.eval(frameTuple, outputTuple);
-          return outputTuple;
+      updateFrameTuple(leftTuple, rightIterator.next());
+
+      if (evalQual()) {
```

Comment:
Except for BNLJoinExec, all join operators (hash, and merge joins) must have join conditions.

BNLJoinExec can be used for cross join, meaning that there is no join condition. Nevertheless, it can have join filter as follows:

```
SELECT x, y, z FROM A, B WHERE concat (x,y) = 'keyword'
```

```concat (x,y) = 'keyword'``` will be play a join filter in BNL join operator. So, we need to consider join condition and filter separately in BNLJoinExec.